File: library_csharp/AddressChangeUnit.cs
namespace CheatEngine.Library;

public static class AddressChangeUnit
{
    public static void ProcessAddress(string address, VariableType variableType, out string resolvedAddress, bool showAsHexadecimal = false, bool showAsSigned = false, int byteSize = 1)
    {
        var resolved = SymbolHandler.Default.GetAddressFromName(address, false, out var hasError);
        resolvedAddress = hasError
            ? "???"
            : ByteInterpreter.ReadAndParseAddress(resolved, variableType, null, showAsHexadecimal, showAsSigned, byteSize);
    }
}

File: library/addresschangeunit.pas
unit AddressChangeUnit;

{$mode objfpc}{$H+}

interface

uses
  Classes, SysUtils, symbolhandler, byteinterpreter, CEFuncProc, LazUTF8;

procedure IProcessAddress(address : WideString ; vartype : TVariableType ; showashexadecimal: Boolean;
  showAsSigned: boolean; bytesize: Integer; out res_address : WideString);stdcall;

implementation

procedure IProcessAddress(address : WideString ; vartype : TVariableType ; showashexadecimal: Boolean;
  showAsSigned: boolean; bytesize: Integer; out res_address : WideString);stdcall;
var a: PtrUInt;
  e: boolean;
  s: string;
begin
  //read the address and display the value it points to

  a:=symhandler.getAddressFromName(utf8toansi(UTF16ToUTF8(address)),false,e);
  if not e then
  begin
    //get the vartype and parse it
    s:=readAndParseAddress(a, vartype,nil,showashexadecimal, showAsSigned, bytesize);
    res_address:=UTF8ToUTF16(s);
  end
  else
    res_address:='???';
end;

end.

